powermon/device.py

```diff
@@ -49,13 +50,19 @@ def __init__(self, name: str, device_id: str = "", model: str = "", manufacturer
         self.manufacturer = manufacturer
         self.port: AbstractPort = port
         self.commands: list[Command] = []
+        self.mqtt_broker = None
 
     def add_command(self, command: Command) -> None:
         """add a command to the devices' list of commands"""
+        log.debug("Adding command: %s", command)
         if command is None:
             return
         # get command definition from protocol
-        command.set_command_definition(self.port.protocol.get_command_with_command_string(command.code))
+        command_definition = self.port.protocol.get_command_with_command_string(command.code)
+        if command_definition is None:
+            log.error("Cannot find command code: %s, in protocol: %s", command.code, self.port.protocol.get_protocol_id())
+            raise RuntimeError(f"Invalid command code: {command.code}, in protocol: {self.port.protocol.get_protocol_id()}")
```

Comment:
i wonder whether its better to raise the error (perhaps even a custom one like InvalidCommandError) in the protocol.get_command_with_command_string function (closer to logic and simpler here), then we can just use `command.set_command_definition(self.port.protocol.get_command_with_command_string(command.code))`